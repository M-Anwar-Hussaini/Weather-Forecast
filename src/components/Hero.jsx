export default function Hero() {
  return (
    <div className="container py-3">
      <div className="row align-items-center">
        <div className="col">
          <img src="afghanistan.png" alt="Afghanistan" className="img-thumbnail" />
        </div>
        <div className="col">
          <h2 className="d-flex flex-column">
            <span>Afghanistan&apos;s</span>
            <span className="text-warning">Weather</span>
            <span>Forecast</span>
          </h2>
        </div>
      </div>
    </div>
  );
}
