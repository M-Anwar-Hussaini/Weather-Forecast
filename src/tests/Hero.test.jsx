import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Hero from '../components/Hero';

describe('Apply tests for Hero component', () => {
  test('Take Snapshot from Hero', () => {
    const home = render(<Hero />);
    expect(home).toMatchSnapshot();
  });

  test('Hero has an image component', () => {
    render(<Hero />);
    const image = screen.getByRole('img');
    expect(image).toBeInTheDocument();
  });

  test('Hero has a heading component', () => {
    render(<Hero />);
    const heading = screen.getByRole('heading');
    expect(heading).toBeInTheDocument();
  });

  test('Heading of Hero has three children', () => {
    render(<Hero />);
    const heading = screen.getByRole('heading');
    expect(heading.children).toHaveLength(3);
  });
});
