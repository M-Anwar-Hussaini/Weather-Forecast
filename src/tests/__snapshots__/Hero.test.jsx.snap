// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Apply tests for Hero component Take Snapshot from Hero 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container py-3"
      >
        <div
          class="row align-items-center"
        >
          <div
            class="col"
          >
            <img
              alt="Afghanistan"
              class="img-thumbnail"
              src="afghanistan.png"
            />
          </div>
          <div
            class="col"
          >
            <h2
              class="d-flex flex-column"
            >
              <span>
                Afghanistan's
              </span>
              <span
                class="text-warning"
              >
                Weather
              </span>
              <span>
                Forecast
              </span>
            </h2>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container py-3"
    >
      <div
        class="row align-items-center"
      >
        <div
          class="col"
        >
          <img
            alt="Afghanistan"
            class="img-thumbnail"
            src="afghanistan.png"
          />
        </div>
        <div
          class="col"
        >
          <h2
            class="d-flex flex-column"
          >
            <span>
              Afghanistan's
            </span>
            <span
              class="text-warning"
            >
              Weather
            </span>
            <span>
              Forecast
            </span>
          </h2>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
